game/scripts/vscripts/libraries/timers.lua

```diff
@@ -98,77 +102,76 @@ function Timers:Think()
 	for _, cb in ipairs(nextTickCallbacks) do
 		DebugCallFunction(cb)
 	end
+	if GameRules:State_Get() >= DOTA_GAMERULES_STATE_POST_GAME then
+		return
+	end
 
 	-- Track game time, since the dt passed in to think is actually wall-clock time not simulation time.
 	local now = GameRules:GetGameTime()
 
-	-- print(table.count(Timers.timers))
 	-- Process timers
-	for k,v in pairs(Timers.timers) do
-		local bUseGameTime = true
-		if v.useGameTime ~= nil and v.useGameTime == false then
-			bUseGameTime = false
-		end
-		local bOldStyle = false
-		if v.useOldStyle ~= nil and v.useOldStyle == true then
-			bOldStyle = true
-		end
+	local timers = Timers.timers
+	self:ExecuteTimers(timers.realTime, Time())
+	self:ExecuteTimers(timers.gameTime, GameRules:GetGameTime())
 
-		local now = GameRules:GetGameTime()
-		if not bUseGameTime then
-			now = Time()
-		end
+	return TIMERS_THINK
+end
 
-		if v.endTime == nil then
-			v.endTime = now
+function Timers:ExecuteTimers(timerList, now)
+	--Empty timer, ignore
+	if not timerList[1] then return end
+	
+	--Timers are alr. sorted by end time upon insertion
+	local currentTimer = timerList[1]
+	
+	local bOldStyle = currentTimer.useOldStyle == true
```

Comment:
I don't think we use old style somewhere, while you're at it could you remove it?